### DISCUSSION TITLE
Request for documentation on cloud-controller-manager component

### DISCUSSION THREAD

**Discussion by daniel-s-palmer** [rancher/rke2#4268] [Q&A] [answered]
This is a follow on to my other ticket here: https://github.com/rancher/rke2/issues/4183

After playing around with the new rke2 version (1.27.1), I am noticing that the cluster automatically includes `cloud-controller-manager`

I am requesting documentation on how this component is recommended to be used with cluster-autoscaler in AWS. Since Kubernetes has moved away from in-tree providers, it is not obvious how one is expected to configure cloud-controller-manager in rke2 to be compatible with AWS.

Am I expected to disable rke2's cloud-controller-manager and deploy the aws cloud provider? https://github.com/kubernetes/cloud-provider-aws

Or is it possible to configure cluster-autoscaler to be compatible with the rke2 cloud-controller-manager in AWS?

Simply setting `--cloud-provider=aws` on the `cloud-controller-manager` pod breaks it (which isn't that surprising)

**Comment by brandond** [comment, marked as answer]
RKE2 uses the same stub cloud-controller-manager as K3s that just sets node internal/external IPs and clears the uninitialized taint. It is intended for use where there is no real cloud provider to integrate with, or where cloud provider integration is not desired. You can find docs on it here: https://docs.k3s.io/networking#deploying-an-external-cloud-controller-manager

If you want to deploy the AWS cloud provider (https://github.com/kubernetes/cloud-provider-aws) you should start RKE2 with --disable-cloud-provider, and deploy cloud-provider-aws instead.

**Reply by daniel-s-palmer** [reply]
I thought that might be the case. Thank you for the clarification!

**Reply by maxisam** [reply]
According to the doc https://docs.k3s.io/networking/networking-services#deploying-an-external-cloud-controller-manager

It seems like if we deploy RKE2 on-prem, we still need to keep CCM.

However, is it ok for us to set cloud-provider-name = "" instead and disable cloud controller ?

This way we don't have the cloud controller at all and the nodes will not remain tainted.

I believe lots of people are doing this way but not sure if this is ok?

**Comment by flphvlck** [comment]
Hi @brandond it would be nice to have some clarification about the cloud-controller-manager component directly in the docs (maybe in https://docs.rke2.io/architecture). I'm new to RKE2 (coming from everything vanilla K8s) and it wasn't clear to me.
First thing which came to my mind was - should I delete it or don't if I'm completely on bare metal?

**Reply by brandond** [reply]
Leave it, unless you like having your nodes come up uninitialized.

**Reply by flphvlck** [reply]
I think you summed it up nicely in https://github.com/rancher/rke2/discussions/4268#discussioncomment-5952737. How about adding your answer to the RKE2 docs?
I can ask for it via PR, but these are your words.
